server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/rfc/SerializerTest.scala

```diff
@@ -0,0 +1,177 @@
+
+object SerializerTest {
+  private val ALGO_1 = "i;ascii-numeric"
+  private val ALGO_2 = "i;ascii-casemap"
+  private val ALGO_3 = "i;unicode-casemap"
+  private val MAX_SIZE_UPLOAD = 50000000
+  private val MAX_CONCURRENT_UPLOAD = 8
+  private val MAX_SIZE_REQUEST = 10000000
+  private val MAX_CONCURRENT_REQUESTS = 10000000
+  private val MAX_CALLS_IN_REQUEST = 32
+  private val MAX_OBJECTS_IN_GET = 256
+  private val MAX_OBJECTS_IN_SET = 128
+  private val USER_1 = "user1@james.org"
+  private val USER_1_ID = Id("user1Id")
+  private val USER_2 = "user2@james.org"
+  private val USER_2_ID = Id("user2Id")
+  private val URL = "http://james.org"
+  private val STATE = "fda9342jcm"
+
+  private val CORE_IDENTIFIER = CapabilityIdentifier(new URI("urn:ietf:params:jmap:core"))
+  private val MAIL_IDENTIFIER = CapabilityIdentifier(new URI("urn:ietf:params:jmap:mail"))
+  private val CONTACT_IDENTIFIER = CapabilityIdentifier(new URI("urn:ietf:params:jmap:contact"))
+
+  private val CORE_CAPABILITY = CoreCapability(
+    maxSizeUpload = MAX_SIZE_UPLOAD, maxConcurrentUpload = MAX_CONCURRENT_UPLOAD,
+    maxSizeRequest = MAX_SIZE_REQUEST, maxConcurrentRequests = MAX_CONCURRENT_REQUESTS,
+    maxCallsInRequest = MAX_CALLS_IN_REQUEST, maxObjectsInGet = MAX_OBJECTS_IN_GET, maxObjectsInSet = MAX_OBJECTS_IN_SET,
+    collationAlgorithms = List(ALGO_1, ALGO_2, ALGO_3))
+  private val CAPABILITIES = Map(
+    CORE_IDENTIFIER -> CORE_CAPABILITY
+  )
+
+  private val ACCOUNT_1 = Account(
+    name = USER_1,
+    isPersonal = true,
+    isReadOnly = false,
+    accountCapabilities = CAPABILITIES)
+  private val ACCOUNT_2 = Account(
+    name = USER_2,
+    isPersonal = false,
+    isReadOnly = false,
+    accountCapabilities = CAPABILITIES)
+  private val ACCOUNTS = Map(
+    USER_1_ID -> ACCOUNT_1,
+    USER_2_ID -> ACCOUNT_2,
+  )
+  private val PRIMARY_ACCOUNTS = Map(
+    MAIL_IDENTIFIER -> USER_1_ID,
+    CONTACT_IDENTIFIER -> USER_2_ID
+  )
+
+  private val SESSION = Session(
+    capabilities = CAPABILITIES,
+    accounts = ACCOUNTS,
+    primaryAccount = PRIMARY_ACCOUNTS,
+    username = USER_1,
+    apiUrl = URL,
+    downloadUrl = URL,
+    uploadUrl = URL,
+    eventSourceUrl = URL,
+    state = STATE)
+
+  private val SESSION_JSON =
+    s"""{
+      |  "capabilities": {
+      |    "urn:ietf:params:jmap:core": {
+      |      "maxSizeUpload": $MAX_SIZE_UPLOAD,
+      |      "maxConcurrentUpload": $MAX_CONCURRENT_UPLOAD,
+      |      "maxSizeRequest": $MAX_SIZE_REQUEST,
+      |      "maxConcurrentRequests": $MAX_CONCURRENT_REQUESTS,
+      |      "maxCallsInRequest": $MAX_CALLS_IN_REQUEST,
+      |      "maxObjectsInGet": $MAX_OBJECTS_IN_GET,
+      |      "maxObjectsInSet": $MAX_OBJECTS_IN_SET,
+      |      "collationAlgorithms": [
+      |        "$ALGO_1",
+      |        "$ALGO_2",
+      |        "$ALGO_3"
+      |      ],
+      |      "_type": "scala.org.apache.james.jmap.rfc.model.CoreCapability"
```

Comment:
Remove this uggly `_type` field everywhere

Reply:
That's what I couldn't do, can someone please suggest?

Reply:
@remk do you know? Scala serialize my object with a concrete type (CoreCapability) while the generic type is Capability. I guess that's reason it put `_type` field. But IDK how to remove it

Reply:
The `_type` field is the discriminator to deserialize sealed hiearachie.
It can be customized see : https://www.playframework.com/documentation/2.8.x/ScalaJsonAutomated#Custom-Naming-Strategies

Reply:
(we don't want to deserialize anything here, especially not with a naming strategy relying on additional object properties.

Reply:
> The _type field is the discriminator to deserialize sealed hiearachie.
It can be customized see

I tried to set the `discriminator` in `JsonConfiguration` to null or empty, it doesn't work. This fix is such a pain